src/lib.rs

```diff
@@ -1,3 +1,66 @@
-#[test]
-fn it_works() {
+mod test;
+
+use std::collections::HashMap;
+
+/// A structure containing all the common attributes
+/// of the basic OSM data types
+/// 
+/// See: [OSM wiki - Common attributes](https://wiki.openstreetmap.org/wiki/Elements#Common_attributes)
+pub struct ElementInfo {
+    pub id: i64,
+    pub user: String,
+    pub uid: i32,
+    pub timestamp: String,
+    pub visible: bool,
+    pub version: i32,
+    pub changeset: i64
 }
+
+/// A node is a single Point holding `lat` and `lon` coordinates (WGS84 reference). 
+/// They are used to define standalone point features. They are also used for modelling 
+/// a Way or they can be a member of a relation.
+/// 
+/// See: [OSM wiki - Node](https://wiki.openstreetmap.org/wiki/Node)
+pub struct Node {
+    pub element_info: ElementInfo,
+    pub tags: HashMap<String, String>,
+    pub lat: f64,
+    pub lon: f64
```

Comment:
These fields can be encoded using i32 for compactness (10^-7 degrees)

Reply:
I went along with the documentation in the wiki which says: 
> Do not use IEEE 32-bit floating point data type since it is limited to about 5 decimal places for the highest longitude.

But it also says: 
> A 32-bit method used by the Rails port is to use an integer (by multiplying each coordinate in degrees by 1E7 and rounding it: this allows to cover all absolute signed coordinates in ±214.7483647 degrees, or a maximum difference of 429.4967295 degrees, a bit more than what is needed).

See: [OSM Wiki - Node](https://wiki.openstreetmap.org/wiki/Node#Structure)